.github/workflows/release.yml

```diff
@@ -29,10 +29,70 @@ permissions:
 jobs:
     release:
         runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                node-version: [18.x]
 
         steps:
             # Check out the repo with credentials that can bypass branch protection, and fetch git history instead of just latest commit
             - uses: actions/checkout@v3
               with:
                   token: ${{ secrets.AUTOMATION_USER_TOKEN }}
                   fetch-depth: 0
+
+            - name: Configure AWS Credentials
+              uses: aws-actions/configure-aws-credentials@v2.2.0
+              with:
+                  aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_GHA }}
+                  aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_GHA }}
+                  aws-region: us-east-1
+
+            - name: Configure git
+              run: |
+                  git config --global user.email "github-tracker-bot@taplytics.com"
+                  git config --global user.name "DevCycle Automation"
+
+            - name: Use Node.js ${{ matrix.node-version }}
+              uses: actions/setup-node@v3
+              with:
+                  node-version: ${{ matrix.node-version }}
+                  cache: 'yarn'
+
+            - run: yarn --immutable
+
+            # Only one of the following two steps will run, depending on whether the `update_all` input is set
+            - name: Increment affected package versions with lerna
+              run: |
+                  ./scripts/lerna-version-ci.sh --version-increment-type "${{ inputs.version-increment-type }}"
+              if: inputs.update_all != true
+
+            - name: Increment all package versions with lerna
+              run: |
+                  ./scripts/lerna-version-ci.sh --all-packages --version-increment-type="${{ inputs.version-increment-type }}"
+              if: inputs.update_all == true
+
+            - name: Push version changes and tags
+              run: |
+                  git push origin HEAD:main --follow-tags
+              if: inputs.draft != true && github.ref == 'refs/heads/main'
+
+            - name: Push to npm
+              run: |
+                  # Use the NPM_TOKEN environment variable to authenticate to the registry
+                  echo 'npmAuthToken: ${NPM_TOKEN}' >> .yarnrc.yml
+
+                  yarn npm-publish
+              env:
+                  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+                  DEVCYCLE_PROD_SLEUTH_API_TOKEN: ${{ secrets.SLEUTH_API_KEY }}
+              if: inputs.draft != true
+
+            - name: Build UMD bundles
+              run: |
+                  yarn nx build:cdn js
+              env:
+                  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+            - name: Upload CDN Release
+              run: aws s3 sync dist/sdk/js s3://js.devcycle.com/ --include "devcycle.min.js*" --acl public-read
+              if: inputs.draft != true && github.ref == 'refs/heads/main'
```

Comment:
We could probably combine these into one step too if needed

Reply:
It's been helpful to have them split for testing that the CDN build works without actually pushing anything.